### ISSUE TITLE
Matrix capsules with EM routing

### ISSUE THREAD

**Issue by changkun** [changkun/applied-ml#2]
- **PDF**: https://openreview.net/pdf?id=HJWLfGWRb
- **Code**: https://github.com/www0wwwjs1/Matrix-Capsules-EM-Tensorflow/ (open-source-impl)
- **Reason**:
  + ICLR 2018 accepted paper, Hinton's paper
  + bring EM algorithm and conducted a better routing mechanism than original Dynamic Routing
  + impair SGD and loss function plays the most important role in a neural network
  + generic definition of Capsule in Deep Learning
- **Related Works**: #1 
- **Comments**:
  + GMM + DL
  + The capsule network is still in the very beginning, and the original accepted paper and its open source code from the author are inconsistent (see [Sarasra/models](https://github.com/Sarasra/models/tree/master/research/capsules), *"no routing is used between Conv1 and PrimaryCapsules"* from paper, but the code uses routing). The most interesting idea from CapsNet its self-explanable and self-evident, it brings the idea of built-in interpretability for building deep models.
  + Material for understanding: capsule blue print, https://docs.google.com/presentation/d/1ccEsDtqjyQVvEvF-MDfwwnXqht5mPRE3c800vs5yJNY/edit?usp=drivesdk
